components/datetime/src/datetime.rs

```diff
@@ -66,15 +70,18 @@ impl<C: CldrCalendar> TimeFormat<C> {
     ///
     /// ```
     /// use icu::calendar::Gregorian;
-    /// use icu::datetime::{TimeFormat, options::length::Time};
+    /// use icu::datetime::{options::length::Time, TimeFormat};
     /// use icu::locid::locale;
-    /// use icu_provider::inv::InvariantDataProvider;
     ///
-    /// let provider = InvariantDataProvider;
-    ///
-    /// let tf = TimeFormat::<Gregorian>::try_new(locale!("en"), &provider, Time::Short, None);
+    /// let provider = icu_testdata::get_provider();
     ///
-    /// assert_eq!(tf.is_ok(), true);
+    /// TimeFormat::<Gregorian>::try_new(
+    ///     locale!("en"),
+    ///     &provider,
+    ///     Time::Short,
+    ///     None,
+    /// )
+    /// .unwrap();
```

Comment:
`nit`: `expect` instead of `unwrap` here and everywhere else in this PR.

Reply:
At least now they're `unwrap`s instead of `.is_ok() == true` so they're easier to find in a later clean-up